bot/processors/pennychat.py

```diff
@@ -0,0 +1,402 @@
+    date_text = f"*Date and Time*\n<!date^{timestamp}^{{date_pretty}} at {{time}}|{penny_chat.date}>"
+
+    save_message = [
+        {
+            "type": "section",
+            "text": {
+                "type": "mrkdwn",
+                "text": f"*New Penny Chat: {penny_chat.title}*"
+            },
+            "accessory": {
+                "type": "button",
+                "text": {
+                    "type": "plain_text",
+                    "text": "Edit Details :pencil2:",
+                    "emoji": True
+                },
+                "action_id": "penny_chat_details"
+            }
+        },
+        {
+            "type": "divider"
+        },
+        {
+            "type": "section",
+            "text": {
+                "type": "mrkdwn",
+                "text": f"*Description*\n{penny_chat.description}"
+            }
+        },
+        {
+            "type": "section",
+            "text": {
+                "type": "mrkdwn",
+                "text": date_text
+            },
+            "accessory": {
+                "type": "button",
+                "text": {
+                    "type": "plain_text",
+                    "text": "Add to Google Calendar :calendar:",
+                    "emoji": True
+                },
+                "url": "https://google.com"
+            }
+        },
+        {
+            "type": "section",
+            "text": {
+                "type": "mrkdwn",
+                "text": f"*Sharing With*\n{share_string}"
+            }
+        },
+        {
+            "type": "actions",
+            "elements": [
+                {
+                    "type": "button",
+                    "text": {
+                        "type": "plain_text",
+                        "text": "Share",
+                        "emoji": True
+                    },
+                    "action_id": "penny_chat_share",
+                    "style": "primary"
+                },
+                {
+                    "type": "button",
+                    "text": {
+                        "type": "plain_text",
+                        "text": "Delete",
+                        "emoji": True
+                    },
+                    "style": "danger"
+                }
+            ]
+        }
+    ]
+    return save_message
+
+
+def shared_message_template(penny_chat):
+    timestamp = int(penny_chat.date.astimezone(utc).timestamp())
+    date_text = f"*Date and Time*\n<!date^{timestamp}^{{date_pretty}} at {{time}}|{penny_chat.date}>"
+
+    start_date = penny_chat.date.astimezone(utc).strftime("%Y%m%dT%H%M%SZ")
+    end_date = (penny_chat.date.astimezone(utc) + timedelta(hours=1)).strftime("%Y%m%dT%H%M%SZ")
+    google_cal_url = f'https://calendar.google.com/calendar/render?action=TEMPLATE&text=' \
+        f'{urllib.parse.quote(penny_chat.title)}&dates=' \
+        f'{start_date}/{end_date}&details=' \
+        f'{urllib.parse.quote(penny_chat.description)}'
```

Comment:
nitpick - split up the lines so that the url params are at the start of each line 
```
google_cal_url = (
    'https://calendar.google.com/calendar/render?'
    'action=TEMPLATE'
    f'text={urllib.parse.quote(penny_chat.title)}&'
    f'dates='{start_date}/{end_date}&'
    f'details={urllib.parse.quote(penny_chat.description)}'
)
```
(commas intentionally missing - it's a weird string concatenation syntax)

Reply:
the [calendar events api](https://developers.google.com/calendar/v3/reference/events) mentions a `hangoutLink` parameter. Maybe see if we can create default google hangout links? up to you

Reply:
It looks like we would need to authenticate with google to do so, which I think is more work than it is worth right now.